op-node/rollup/driver/state.go

```diff
@@ -426,34 +427,22 @@ func (s *Driver) syncStatus() *eth.SyncStatus {
 	}
 }
 
-// SyncStatus blocks the driver event loop and captures the syncing status.
-// If the event loop is too busy and the context expires, a context error is returned.
+// updateSyncStatusCache updates the cached sync status
+func (s *Driver) updateSyncStatusCache() {
+	ss := s.syncStatus()
+	s.syncStatusCache.Store(ss)
+}
+
+// SyncStatus returns the syncing status.
 func (s *Driver) SyncStatus(ctx context.Context) (*eth.SyncStatus, error) {
-	wait := make(chan struct{})
-	select {
-	case s.stateReq <- wait:
-		resp := s.syncStatus()
-		<-wait
-		return resp, nil
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	}
+	return s.syncStatusCache.Load(), nil
 }
 
-// BlockRefWithStatus blocks the driver event loop and captures the syncing status,
-// along with an L2 block reference by number consistent with that same status.
-// If the event loop is too busy and the context expires, a context error is returned.
+// BlockRefWithStatus returns the syncing status along with an L2 block reference by number consistent with that same status.
```

Comment:
I believe that this should be blocking for the main loop. We rely on the invariant that the L2 block by number is part of the same chain as the sync status.

Reply:
Wasn't sure if we should maintain this behavior. It's a weak invariant because a reorg could in theory occur while the main loop is blocked. I can revert back to the way it was though. Main motivation for this is to prevent the batcher, for example, from being blocked waiting the sync status. Updating `BlockRefWithStatus` is just a bonus.

Reply:
If anything, I think we need to increase the strength of this invariant for the proposer. Having an L1 reorg so the L1 state is out of sync with the L2 state is fine. The biggest worry is if the L1 reorg has propagated to the state loop, but that L2 state has not been updated.

Reply:
Sorry for the delay. I've reverted this function back to what it was. It avoids desync L2 state and block info desync.